### PR TITLE
Support tokenless authentication

### PR DESCRIPTION
**Pull request by shuheiktgw**
Updated CLI to support tokenless authentication. 

In order to achieve that, I've mainly focused on refactoring utils/http_client.py. Previously, the command classes decide how to access API but I found it redundant so I changed http_client.py and remove some duplicate codes.

### CODE REVIEW

File: launchable/utils/authentication.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        return {
+            'GitHub-Actions': os.environ['GITHUB_ACTIONS'],
+            'GitHub-Run-Id': os.environ['GITHUB_RUN_ID'],
+            'GitHub-Repository': os.environ['GITHUB_REPOSITORY'],
+            'GitHub-Workflow': os.environ['GITHUB_WORKFLOW'],
+            'GitHub-Run-Number': os.environ['GITHUB_RUN_NUMBER'],
+            'GitHub-Event-Name': os.environ['GITHUB_EVENT_NAME'],
+            # GITHUB_PR_HEAD_SHA might not exist
+            'GitHub-Pr-Head-Sha': os.getenv('GITHUB_PR_HEAD_SHA'),
+            'GitHub-Sha': os.environ['GITHUB_SHA'],
```
Comment (Konboi) [inline launchable/utils/authentication.py:33]: Memo: https://docs.github.com/en/actions/reference/environment-variables

File: launchable/commands/record/tests.py [hunk 2]
```diff
@@ -174,95 +173,54 @@ def scan(self, base: str, pattern: str):
...
+                exs = []
                 for report in reports:
                     try:
-                        yield from self.parse_func(report)
-
-                    except Exception as e:
-                        exceptions.append(Exception(
-                            "Failed to process a report file: {}".format(report), e))
+                        for c in self.parse_func(report):
+                            cases.append(c)
```
[the remark this answers is not in the document]
Comment (shuheiktgw) [reply launchable/commands/record/tests.py]: Fixed it!
- 718a569

File: launchable/utils/authentication.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            return org, workspace
+        except ValueError:
+            return None, None
+
+    return os.getenv(ORGANIZATION_KEY), os.getenv(WORKSPACE_KEY)
+
+
+def authentication_headers():
+    token = os.getenv(TOKEN_KEY)
+    if token is not None:
```
[the remark this answers is not in the document]
Comment (shuheiktgw) [reply launchable/utils/authentication.py]: Fixed this too

- fc91bae

**Comment by shuheiktgw** [comment]
@Konboi I believe I've fixed it up so can you review the PR again?

**Review by Konboi — APPROVED** [review]
LGTM 🚀

**Comment by shuheiktgw** [comment]
Thank you!

**Pull request merged**